src/Api.Rest/Common/Client/ObjectSerializer.cs

```diff
@@ -0,0 +1,190 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2022                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.Common.Client
+{
+	using System;
+	using System.Collections.Generic;
+	using System.IO;
+	using System.Text;
+	using JetBrains.Annotations;
+	using Zeiss.PiWeb.Api.Rest.Dtos.Converter;
+	using Zeiss.PiWeb.Api.Rest.Dtos.JsonConverters;
+
+	/// <summary>
+	/// Provides a set of static extensions to <see cref="IObjectSerializer"/>.
+	/// </summary>
+	public static class ObjectSerializer
+	{
+		#region members
+
+		/// <summary>
+		/// Newtonsoft.Json de-/serialization
+		/// </summary>
+		public static readonly IObjectSerializer NewtonsoftJson = new NewtonsoftJsonSerializer();
+
+		/// <summary>
+		/// System.Text.Json de-/serialization
+		/// </summary>
+		public static readonly IObjectSerializer SystemTextJson = new SystemTextJsonSerializer();
+
+		/// <summary>
+		/// Default de-/serialization
+		/// </summary>
+		public static readonly IObjectSerializer Default = NewtonsoftJson;
+
+		#endregion
+
+		#region methods
+
+		internal static IEnumerable<T> DeserializeEnumeratedObject<T>( [NotNull] this IObjectSerializer serializer, [NotNull] Stream data )
+		{
+			if( serializer == null ) throw new ArgumentNullException( nameof( serializer ) );
+			if( data == null ) throw new ArgumentNullException( nameof( data ) );
+
+			IEnumerable<T> DeserializeEnumeratedObject()

```

Comment:
Is this local function needed here?

Reply:
It is not needed, but very convenient. The `ArgumentException` will be thrown immediately on calling the method, otherwise the exception will not be thrown until the enumeration of the returned sequence begins.

See https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions#local-functions-and-exceptions